### PR TITLE
optimize server startup parameters

### LINKED ISSUE #1100: eureka registry support customize IP
when i deploy seata on cloud host，can not get real ip
-----------------------------------------------------------------------
2019-05-24 09:21:26.491 WARN [main]io.seata.common.util.NetUtil.getLocalAddress0:155 -Failed to retrieving ip address, ecs-bd21: ecs-bd21: Name or service not known
java.net.UnknownHostException: ecs-bd21: ecs-bd21: Name or service not known
at java.net.InetAddress.getLocalHost(InetAddress.java:1506)
at io.seata.common.util.NetUtil.getLocalAddress0(NetUtil.java:150)
at io.seata.common.util.NetUtil.getLocalAddress(NetUtil.java:142)
at io.seata.common.util.NetUtil.getLocalIp(NetUtil.java:119)
at io.seata.server.Server.main(Server.java:87)
Caused by: java.net.UnknownHostException: ecs-bd21: Name or service not known
at java.net.Inet6AddressImpl.lookupAllHostAddr(Native Method)
at java.net.InetAddress$2.lookupAllHostAddr(InetAddress.java:929)
at java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1324)
at java.net.InetAddress.getLocalHost(InetAddress.java:1501)
... 4 common frames omitted

### PR DESCRIPTION
**Pull request by xingfudeshi**
### Ⅰ. Describe what this PR did
1.optimize server startup parameters,makes parameter passing more flexible.
2.support bind specific host,default will bind 0.0.0.0.
`sh seata-server.sh --help`
image

### Ⅱ. Does this pull request fix one issue?

fixes #1100 

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
unit test provided,and the integration test passed.

### CODE REVIEW

File: server/src/main/java/io/seata/server/ParameterParser.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    private static final int SERVER_DEFAULT_PORT = 8091;
+    private static final String SERVER_DEFAULT_BIND_IP = "0.0.0.0";
+    private static final String SERVER_DEFAULT_STORE_MODE = "file";
+
+
+    @Parameter(names = "--help", help = true)
+    private boolean help;
+    @Parameter(names = {"--host", "-h"}, description = "The host to bind.", order = 1)
+    private String host = SERVER_DEFAULT_BIND_IP;
+    @Parameter(names = {"--port", "-p"}, description = "The port to listen.", order = 2)
```
Comment (slievrly) [inline server/src/main/java/io/seata/server/ParameterParser.java]: The order of the parameters has changed, and the user may not start up using the old command.

File: server/src/main/java/io/seata/server/ParameterParser.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+/**
+ * The type parameter parser
+ *
+ * @author xingfudeshi@gmail.com
+ * @date 2019/05/30
+ */
+public class ParameterParser {
+    private static final String PROGRAM_NAME = "sh seata-server.sh";
+    private static final int SERVER_DEFAULT_PORT = 8091;
+    private static final String SERVER_DEFAULT_BIND_IP = "0.0.0.0";
```
Comment (slievrly) [inline server/src/main/java/io/seata/server/ParameterParser.java]: Here I want to bind an external ip to 127 and 0.0.0.0 even if the user does not pass this parameter. If the user does not pass, the returned xid is 0.0.0.0:8091:xxxxx, and I want to see the ip address of the destination server.

File: server/src/main/java/io/seata/server/ParameterParser.java (hunk 1 above)
Comment (xingfudeshi) [reply server/src/main/java/io/seata/server/ParameterParser.java]: > The order of the parameters has changed, and the user may not start up using the old command.

Do you mean the order attribute in the @Parameter?It's only the order for displaying of each option.not the parameter passing order.
when type seata-server --help,will display result as following:
image

File: server/src/main/java/io/seata/server/ParameterParser.java (hunk 2 above)
Comment (xingfudeshi) [reply server/src/main/java/io/seata/server/ParameterParser.java]: > Here I want to bind an external ip not 127 and 0.0.0.0 even if the user does not pass this parameter. If the user does not pass, the returned xid is 0.0.0.0:8091:xxxxx, and I want to see the ip address of the destination server.

Got it,will change it.

File: server/src/main/java/io/seata/server/ParameterParser.java (hunk 1 above)
Comment (slievrly) [reply server/src/main/java/io/seata/server/ParameterParser.java]: > > The order of the parameters has changed, and the user may not start up using the old command.
> 
> Do you mean the order attribute in the @parameter?It's only the order for displaying of each option.not the parameter passing order.
> when type seata-server --help,will display result as following:
> image

ok, I misunderstand.

File: server/src/main/java/io/seata/server/ParameterParser.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+import com.beust.jcommander.ParameterException;
+
+/**
+ * The type parameter parser
+ *
+ * @author xingfudeshi@gmail.com
+ * @date 2019/05/30
+ */
+public class ParameterParser {
+    private static final String PROGRAM_NAME = "sh seata-server.sh";
```
Comment (leizhiyuan) [inline server/src/main/java/io/seata/server/ParameterParser.java]: what for windows？

Comment (xingfudeshi) [reply server/src/main/java/io/seata/server/ParameterParser.java]: > what for windows？

I think we can simply write seata-server,and the final result will be like:
image
in addition,it's not necessary to detect the OS for this.

File: server/src/main/java/io/seata/server/ParameterParser.java (hunk 2 above)
Comment (xingfudeshi) [reply server/src/main/java/io/seata/server/ParameterParser.java]: > Here I want to bind an external ip not 127 and 0.0.0.0 even if the user does not pass this parameter. If the user does not pass, the returned xid is 0.0.0.0:8091:xxxxx, and I want to see the ip address of the destination server.

it's done.

File: common/src/main/java/io/seata/common/util/NetUtil.java [hunk 4]
```diff
@@ -196,12 +196,33 @@ public static void validAddress(InetSocketAddress address) {
...
+     * @return true if the given address is valid
+     */
     private static boolean isValidAddress(InetAddress address) {
         if (address == null || address.isLoopbackAddress()) {
             return false;
         }
         String name = address.getHostAddress();
-        return (name != null && !ANY_HOST.equals(name) && !LOCALHOST.equals(name) && IP_PATTERN.matcher(name)
-            .matches());
+        return isValidIp(name, false);
```
Comment (slievrly) [inline common/src/main/java/io/seata/common/util/NetUtil.java]: name->ip

File: server/src/main/java/io/seata/server/ParameterParser.java (hunk 3 above)
Comment (slievrly) [reply server/src/main/java/io/seata/server/ParameterParser.java]: May be sh seata-server.sh(for linux and mac) or cmd seata-server.bat(for windows) [options]

File: common/src/main/java/io/seata/common/util/NetUtil.java (hunk 4 above)
Comment (xingfudeshi) [reply common/src/main/java/io/seata/common/util/NetUtil.java]: ok.i will fix it.

File: server/src/main/java/io/seata/server/ParameterParser.java (hunk 3 above)
Comment (xingfudeshi) [reply server/src/main/java/io/seata/server/ParameterParser.java]: ok.

Comment (xingfudeshi) [reply server/src/main/java/io/seata/server/ParameterParser.java]: image

**Pull request merged**